### ISSUE TITLE
`param-decorator-name-matches-route-param` skips all routes if controller contains a variable

### ISSUE THREAD

**Issue by ChrisMBarr** [darraghoriordan/eslint-plugin-nestjs-typed#296]
In our app we don't use route params in controllers, but we do save these paths in a shared folder for our monorepo.
If a controller contains a variable, then the `param-decorator-name-matches-route-param` rule will not check any routes at all.

Here's a quick example: This report a failure as we all would expect

```ts
@Controller('api/v2/example')
export class ExampleController {
  @Get('test/:id')
  public getProjectByStringAndDate$(@Param('iidd') id: string) {
 //                                 ^^^^^^^^^^^^^^ Failure reported here, as expected
    console.log(id);
  }
}
```
However this reports no failures at all!

```ts
import { exampleBaseRoute } from '@monorepo-shared/routes';

@Controller(exampleBaseRoute)
export class ExampleController {
  @Get('test/:id')
  public getProjectByStringAndDate$(@Param('iidd') id: string) {
    console.log(id);
  }
}
```
